zerver/views/home.py

```diff
@@ -168,6 +169,22 @@ def home_real(request: HttpRequest) -> HttpResponse:
         user_profile = None
         realm = get_valid_realm_from_request(request)
 
+        # TODO: Allow user to directly login as web_public_visitor
+        # if `next` contains a web-public-stream.
```

Comment:
We probably want this behavior conditional on the realm having web public streams enabled?

Reply:
Yeah, I am not sure what is the best way to do that. We can query all the streams for if any of them are web public or do we plan to have a separate setting for it. I imagine the setting could be useful to orgs if they want to disable web public view temporarily.

Reply:
https://chat.zulip.org/#narrow/stream/2-general/topic/web.20public.20view.20setting